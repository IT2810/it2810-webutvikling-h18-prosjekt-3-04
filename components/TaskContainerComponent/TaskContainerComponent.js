import React, {Component} from 'react';
import {View, Text, TouchableHighlight, Image} from 'react-native';
import styles from "./styles/styles";
import CheckboxComponent from '../CheckboxComponent/CheckboxComponent';
import PropTypes from 'prop-types';

export default class TaskContainerComponent extends Component{

    constructor() {
        super();
        this.onPress = this.onPress.bind(this);
        this.state= {
            checked: false,
        };

    }

    static propTypes = {
        type: PropTypes.string,
        data: PropTypes.string,
        deadline: PropTypes.string,
        onPress: PropTypes.func,
        isChecked: PropTypes.bool,
    };
    onPress = () => {
        this.setState({ checked: !this.state.checked });
        this.props.isChecked = this.state.checked;
    };


    render(){
        return (
            <TouchableHighlight underlayColor={"rgba(0,0,0,0)"} onPress = { this.onPress } >
                {this.props.type === "motivational" ? <View style={styles.motivationalElement}>
                        <Text style={styles.motivationalText}>{this.props.data}</Text>
                    </View> :
                    <View style={styles.taskObject}>
                        <View style={styles.textFlex}>
                            {this.props.type === "image" ? <Image style={styles.image} source={{uri: 'data:image/png;base64,' + this.props.data}}/>
                                    : <Text style={styles.taskText}>{this.props.data}</Text>}
                            <Text style={styles.dateStyle}>{this.props.deadline}</Text>
                        </View>
                        <View style={styles.checkFlex}>
                            <CheckboxComponent checked={this.state.checked} onPress={this.onPress}/>
                        </View>
                    </View>
                }

            </TouchableHighlight>
        );
    }
}

