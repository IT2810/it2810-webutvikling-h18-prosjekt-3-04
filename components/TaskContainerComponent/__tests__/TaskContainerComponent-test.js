import React from 'react';
import TaskContainerComponent from '../TaskContainerComponent';

import renderer from 'react-test-renderer';


test('renders correctly', () => {
    const tree = renderer.create(<TaskContainerComponent/>).toJSON();
    expect(tree).toMatchSnapshot();
});
