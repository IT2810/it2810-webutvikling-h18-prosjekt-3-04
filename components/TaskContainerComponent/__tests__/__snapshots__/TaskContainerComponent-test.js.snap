// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with type = image 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
>
  <View
    style={
      Object {
        "backgroundColor": "#FFF",
        "borderRadius": 20,
        "elevation": 1,
        "flex": 0,
        "flexDirection": "row",
        "flexWrap": "wrap",
        "height": "auto",
        "justifyContent": "space-between",
        "marginBottom": 0,
        "marginLeft": 10,
        "marginRight": 10,
        "marginTop": 10,
        "padding": 20,
        "shadowColor": "black",
        "shadowOffset": Object {
          "height": 2,
        },
        "shadowOpacity": 0.1,
        "width": 710,
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "center",
          "width": 200,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "data:image/jpeg;base64,undefined",
          }
        }
        style={
          Object {
            "alignSelf": "center",
            "aspectRatio": 1,
            "borderColor": "#f3f3f3",
            "borderRadius": 10,
            "borderWidth": 2,
            "marginLeft": -30,
            "width": 170,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#FF0040",
            "marginTop": 10,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 1,
          "justifyContent": "center",
          "width": 40,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFF",
              "borderColor": "#bcbcbc",
              "borderRadius": 10,
              "borderWidth": 2,
              "height": 40,
              "justifyContent": "center",
              "width": 40,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`renders correctly with type = image 2`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
>
  <View
    style={
      Object {
        "backgroundColor": "#FFF",
        "borderRadius": 20,
        "elevation": 1,
        "flex": 0,
        "flexDirection": "row",
        "flexWrap": "wrap",
        "height": "auto",
        "justifyContent": "space-between",
        "marginBottom": 0,
        "marginLeft": 10,
        "marginRight": 10,
        "marginTop": 10,
        "padding": 20,
        "shadowColor": "black",
        "shadowOffset": Object {
          "height": 2,
        },
        "shadowOpacity": 0.1,
        "width": 710,
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "center",
          "width": 200,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 17,
            "fontWeight": "bold",
          }
        }
      >
        🏃🏻‍♂️ 
        0
         of 
        0
         steps
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#FF0040",
            "marginTop": 10,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 1,
          "justifyContent": "center",
          "width": 40,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFF",
              "borderColor": "#bcbcbc",
              "borderRadius": 10,
              "borderWidth": 2,
              "height": 40,
              "justifyContent": "center",
              "width": 40,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`renders correctly with type = text 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
>
  <View
    style={
      Object {
        "backgroundColor": "#FFF",
        "borderRadius": 20,
        "elevation": 1,
        "flex": 0,
        "flexDirection": "row",
        "flexWrap": "wrap",
        "height": "auto",
        "justifyContent": "space-between",
        "marginBottom": 0,
        "marginLeft": 10,
        "marginRight": 10,
        "marginTop": 10,
        "padding": 20,
        "shadowColor": "black",
        "shadowOffset": Object {
          "height": 2,
        },
        "shadowOpacity": 0.1,
        "width": 710,
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "center",
          "width": 200,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 17,
            "fontWeight": "bold",
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#FF0040",
            "marginTop": 10,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 1,
          "justifyContent": "center",
          "width": 40,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFF",
              "borderColor": "#bcbcbc",
              "borderRadius": 10,
              "borderWidth": 2,
              "height": 40,
              "justifyContent": "center",
              "width": 40,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
